domain/src/main/java/org/oppia/domain/analytics/AnalyticsController.kt

```diff
@@ -0,0 +1,126 @@
+package org.oppia.domain.analytics
+
+import android.content.Context
+import org.oppia.app.model.EventLog
+import org.oppia.app.model.EventLog.EventAction
+import org.oppia.app.model.EventLog.Priority
+import org.oppia.util.logging.EventLogger
+import javax.inject.Inject
+
+const val TEST_TIMESTAMP = 1556094120000
+const val TEST_TOPIC_ID = "test_topicId"
+const val TEST_STORY_ID = "test_storyId"
+const val TEST_EXPLORATION_ID = "test_explorationId"
+const val TEST_QUESTION_ID = "test_questionId"
+
+/**
+ * Controller for handling analytics event logging.
+ * [logTransitionEvent] should be used to log screen transition events.
+ * [logClickEvent] should be used to log button click events.
+ */
+class AnalyticsController @Inject constructor(
+  private val eventLogger: EventLogger
+) {
+  /**
+   * Logs transition events.
+   * These events are given HIGH priority.
+   */
+  fun logTransitionEvent(
+    context: Context,
+    timestamp: Long,
+    eventAction: EventAction,
+    topicId: String?,
+    storyId: String?,
+    explorationId: String?,
+    questionId: String?
+  ) {
+    eventLogger.logEvent(
+      context,
+      createEventLog(
+        timestamp, topicId, eventAction, storyId, explorationId, questionId, Priority.ESSENTIAL
+      )
+    )
+  }
+
+  /**
+   * Logs click events.
+   * These events are given LOW priority.
+   */
+  fun logClickEvent(
+    context: Context,
+    timestamp: Long,
+    eventAction: EventAction,
+    topicId: String?,
+    storyId: String?,
+    explorationId: String?,
+    questionId: String?
+  ) {
+    eventLogger.logEvent(
+      context,
+      createEventLog(
+        timestamp, topicId, eventAction, storyId, explorationId, questionId, Priority.OPTIONAL
+      )
+    )
+  }
+
+  /** Returns an event log containing relevant data for event reporting. */
+  private fun createEventLog(
+    timestamp: Long,
+    topicId: String?,
+    eventAction: EventAction,
+    storyId: String?,
+    explorationId: String?,
+    questionId: String?,
+    priority: Priority
+  ): EventLog {
+
+    return EventLog.newBuilder()
+      .setTimestamp(timestamp)
+      .setActionName(eventAction)
+      .setPriority(priority)
+      .setContext(contextCreation(topicId, storyId, explorationId, questionId))
+      .build()
+  }
+
+  /** Returns the activity context of an event. */
+  private fun contextCreation(
+    topicId: String?,
+    storyId: String?,
+    explorationId: String?,
+    questionId: String?
+  ): EventLog.Context {
+    return if (questionId != null) {
```

Comment:
Based on this, we are creating question context even in cases where the questionId, explorationId, storyId, and topicId are set. 
Maybe we want to separate this function into two:
`createExplorationContext` and `createQuestionContext` , each having different parameters. We can make these functions public so callers can create the correct context and pass the context into logTransition or logClick. What do you think?

Reply:
I think we can make amends to our existing logTransition and logClickEvent functions by splitting them into a total of 4 functions. Like - 
1. `logTransitionWithExplorationContext`
2. `logTransitionWithQuestionContext`
3. `logClickWithExplorationContext`
4. `logClickWithQuestionContext`

and then control their parameters to fulfil the context requirements. Thoughts?

Reply:
Okay yeah I like that idea too, makes sense

Reply:
Actually on second thought, lets not do this. Because: We need a 'no context' scenario and we may add newer contexts in the future. If we add a new context we would need to add 2 more log functions. That's a lot of duplicate code just to change context so lets have a create exploration / create question / create no context situations.

Reply:
Okay, makes sense.

Reply:
Done. PTAL :)